### PR TITLE
fix: change all occurrences of hard-coded ip address '127.0.0.1' to 'localhost'

### PR DESCRIPTION
**Pull request by vraravam**
#### Pre-flight Checklist

1. Please remember that if you are logging a bug for some service that has *stopped working* or is *working incorrectly*, please log the bug [here](https://github.com/getferdi/recipes/issues)
2. If you are requesting support for a **new service** in Ferdi, please log it [here](https://github.com/getferdi/recipes/pulls)
3. Please remember to read the [self-help documentation](https://github.com/getferdi/ferdi#troubleshooting-recipes-self-help) - in case it helps you unblock yourself for issues related to older versions of recipes that were installed on your machine. (These will get automatically upgraded when you upgrade to the newer versions of Ferdi, but to get new recipes between Ferdi releases, this documentation is quite useful.)
4. Please consider supporting Ferdi!
  👉  https://github.com/sponsors/getferdi
  👉  https://opencollective.com/getferdi/donate
5. Please ensure you've completed all of the following.
- [x] I have read the [Contributing Guidelines](https://github.com/getferdi/ferdi/blob/develop/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/getferdi/ferdi/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://github.com/getferdi/ferdi/issues) for a feature request that matches the one I want to file, without success.

#### Description of Change
Change all occurrences of hard-coded ip address '127.0.0.1' to 'localhost'
Also refactored to make this a constant defined in a single file where it is imported for all other
uses.

#### Motivation and Context
Bug #1253 

#### Checklist

- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`npm run prepare-code`)
- [x] `npm test` passes
- [x] I tested/previewed my changes locally

### CODE REVIEW

File: src/config.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ import ms from 'ms';
 
 export const CHECK_INTERVAL = ms('1h'); // How often should we perform checks
 
+export const LOCAL_HOSTNAME = 'localhost';
 export const LOCAL_API = 'http://localhost:3000';
```
Comment (mhatvan) [inline src/config.js:8]: Port here should also be dynamic in the future.

File: src/internal-server/config/session.js [hunk 2]
```diff
@@ -88,7 +88,7 @@ module.exports = {
   |
   */
   redis: {
-    host: '127.0.0.1',
+    host: 'localhost',
     port: 6379,
```
Comment (mhatvan) [inline src/internal-server/config/session.js:92]: Port here should also be dynamic in the future.

**Review by mhatvan — APPROVED** [review]

File: src/config.js (hunk 1 above)
Comment (vraravam) [reply src/config.js:8]: Agreed - the intent of this PR though is to make it into a single location. We can tackle this as part of a different PR.

File: src/internal-server/config/session.js (hunk 2 above)
Comment (vraravam) [reply src/internal-server/config/session.js:92]: Agreed - the intent of this PR though is to make it into a single location. We can tackle this as part of a different PR.

File: src/config.js (hunk 1 above)
Comment (mhatvan) [reply src/config.js:8]: Yeah, thats what I meant to keep it in mind. Ready to merge!

File: src/internal-server/config/session.js (hunk 2 above)
Comment (mhatvan) [reply src/internal-server/config/session.js:92]: Yeah, thats what I meant to keep it in mind. Ready to merge!

**Pull request merged**